src/tools/rbd_wnbd/rbd_wnbd.h

```diff
@@ -69,6 +70,9 @@ struct Config {
   bool remap_failure_fatal = false;
   bool adapter_monitoring_enabled = false;
 
+  bool wait_disk_online = false;
+  int wait_online_timeout = DEFAULT_IMAGE_MAP_TIMEOUT;
```

Comment:
... it also seems like there shouldn't be a timeout on this step.  I see that there are some existing cumulative timeouts in service mode, but it doesn't seem like the plain `map <imagespec>` command has any.  Can the "wait online" step be covered by some existing timeout in service mode, leaving the plain command as is?

Reply:
Right, adding yet another timeout option might seem redundant. We do need to have a timeout though, we can't poll indefinitely. What we could do would be to substract the time it took to issue the map request from the configured map timeout.

We're currently using map timeouts in a few places:

* when the map command issues the request to the centralized service through a named pipe: https://github.com/ceph/ceph/blob/4af9db9102780c7eef05d154f1148c682c8e1603/src/tools/rbd_wnbd/rbd_wnbd.cc#L300-L307
* when the centralized service spawns the map subprocess: https://github.com/ceph/ceph/blob/4af9db9102780c7eef05d154f1148c682c8e1603/src/tools/rbd_wnbd/rbd_wnbd.cc#L785
* when the service restarts: https://github.com/ceph/ceph/blob/4af9db9102780c7eef05d154f1148c682c8e1603/src/tools/rbd_wnbd/rbd_wnbd.cc#L584-L587

The problem is that in the first two situations we're using the default hard coded value instead of honoring the configurable value. There's a TODO, can't remember if there was a specific reason why we kept using the hardcoded value.

The "wait online" step can be performed either on the caller side (as proposed by this patch) and/or on the service side. I'm a bit concerned about performing the wait when the service starts. For example, the service start time may increase significantly if we wait for 100 disks.

I'd revisit the timeout handling after we merge this: https://github.com/ceph/ceph/pull/52540.

Reply:
I've updated the PR, dropping the new parameters and performing the wait on the service side.

Reply:
> On the other hand, it might be the right thing to do, guaranteeing that all rbd mappings are available once the service is running.

Yeah, it's why the service is there after all.

I'm not opposed to optionally skipping the wait, but I strongly believe that the default behavior should be to wait -- especially for a case of mapping a single disk (e.g. what fsx does).